### DISCUSSION TITLE
Can you wrap a full page with q-scroll-area?

### DISCUSSION THREAD

**Discussion by b0ot** [quasarframework/quasar#15916] [General - Components / Directives / etc] [answered]
I need to have the scrollbar visible on desktop and mobile for my application as there are certain elements that don't respond to touch (by design).

I was hoping I could simply wrap the entire page with a q-scroll-area that is always visible, however since you need specify a fixed height/width this seems impossible unless you manually calculate the heights of all elements on the page.

Is there any other solution?

**Comment by suleimanas** [comment]
one way is to use CSS Positioning Classes

`<q-page class="relative-position">`

`<q-scroll-area class="absolute-full">`

**Reply by b0ot** [reply]
This solution almost worked for me!
For whatever reason when I go to a narrow view, the spacing gets odd.

On desktop without the q-page / q-scroll area:
(Everything is centered)
![image](https://github.com/quasarframework/quasar/assets/9140903/e1748865-6fc8-4026-a1a5-81fe888131e3)

When I use the q-page / q-scroll area:
![image](https://github.com/quasarframework/quasar/assets/9140903/af50ce96-d06c-40c9-8b8f-121dff1f547a)
The spacing gets messed up.

**Reply by suleimanas** [reply]
you need to wrap it inside another div so you can center it correctly: q-page>q-scroll-area>div>your_page
maybe if you can make a codepen I can help you with it

**Comment by metalsadman** [comment, marked as answer]
rough example https://codepen.io/metalsadman/pen/YzEoWNa

**Reply by suleimanas** [reply]
perfect example

**Comment by b0ot** [comment]
Is there a way for a component to take as much space as needed up to a specified height and then turn into a scroll-area?

**Comment by marvin-wtt** [comment]
This worked best for me:

```html
// MainLayout.vue
  <q-layout>
    <!-- q-header -->

    <q-scroll-area style="width: 100vw; height: 100vh">
      <div style="max-width: 100vw">
        <q-page-container>
          <router-view />
        </q-page-container>
      </div>
    </q-scroll-area>
    
   </q-layout>
```
